packages/hash/frontend/src/blocks/page/BlockContextMenu/BlockListMenuContent.tsx

```diff
@@ -0,0 +1,101 @@
+import { faSearch } from "@fortawesome/free-solid-svg-icons";
+import {
+  Box,
+  InputAdornment,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  MenuList,
+  TextField,
+  Typography,
+} from "@mui/material";
+import { PopupState } from "material-ui-popup-state/core";
+import { useEffect, useRef, useState, VFC } from "react";
+import {
+  BlockSuggesterProps,
+  getVariantIcon,
+} from "../createSuggester/BlockSuggester";
+import { useFilteredBlocks } from "../createSuggester/useFilteredBlocks";
+import { useUserBlocks } from "../../userBlocks";
+import { FontAwesomeIcon } from "../../../shared/icons";
+
+type BlockListMenuContentProps = {
+  popupState?: PopupState;
+  blockSuggesterProps: BlockSuggesterProps;
+};
+
+export const BlockListMenuContent: VFC<BlockListMenuContentProps> = ({
+  blockSuggesterProps,
+  popupState,
+}) => {
+  const [searchQuery, setSearchQuery] = useState("");
+  const searchInputRef = useRef<HTMLInputElement>(null);
+  const { value: userBlocks } = useUserBlocks();
+  const blocks = useFilteredBlocks(searchQuery, userBlocks);
+
+  useEffect(() => {
+    if (popupState?.isOpen) {
+      searchInputRef.current?.focus();
+    }
+  }, [popupState]);
```

Comment:
This effect prevents switching to some blocks, e.g. Embed, Table. The effect fires and focuses the input, stopping the block changing. I'm not sure why this happens for some blocks but not others.

Reply:
Will look into this

Reply:
I couldn't figure out what could possible be wrong. Commented it out for now in [c867f5c](https://github.com/hashintel/hash/pull/480/commits/c867f5c506c6ee0ccff8531d8a3ecff0f2718405) so that block switching still works. Also added a todo comment. I will address it properly when working on [nested menus](https://app.asana.com/0/1202055186269230/1202128324698413)